mollie/api/client.py

```diff
@@ -179,25 +187,81 @@ def perform_http_call(self, http_method, path, data=None, params=None):
-                http_method, url,
-                verify=True,
-                headers={
-                    'Accept': 'application/json',
-                    'Authorization': 'Bearer {api_key}'.format(api_key=self.api_key),
-                    'Content-Type': 'application/json',
-                    'User-Agent': self.user_agent,
-                    'X-Mollie-Client-Info': self.UNAME,
-                },
-                params=params,
-                data=data,
-                timeout=self.timeout,
-            )
-        except Exception as err:
-            raise RequestError('Unable to communicate with Mollie: {error}'.format(error=err))
+        if self.oauth:
+            try:
+                response = self.oauth.request(
+                    http_method,
+                    url,
+                    params=params,
+                    data=data,
+                )
+            except Exception as err:
+                raise RequestError('Unable to communicate with Mollie: {error}'.format(error=err))
+        else:
+            try:
+                response = requests.request(
+                    http_method, url,
+                    verify=True,
+                    headers={
+                        'Accept': 'application/json',
+                        'Authorization': 'Bearer {api_key}'.format(api_key=self.api_key),
+                        'Content-Type': 'application/json',
+                        'User-Agent': self.user_agent,
+                        'X-Mollie-Client-Info': self.UNAME,
+                    },
+                    params=params,
+                    data=data,
+                    timeout=self.timeout,
+                )
+            except Exception as err:
+                raise RequestError('Unable to communicate with Mollie: {error}'.format(error=err))
         return response
 
+    def setup_oauth(self, client_id, client_secret, redirect_uri, scope, token, set_token):
+        """
+        :param client_id: (string)
+        :param client_secret: (string)
+        :param redirect_uri: (string)
+        :param scope: Mollie connect permissions (list)
+        :param token: The stored token (dict)
+        :param set_token: Callable that stores a token (dict)
+        :return: authorization url (url)
+        """
+        self.set_token = set_token
+        self.client_secret = client_secret
+        self.oauth = OAuth2Session(
+            client_id,
+            auto_refresh_kwargs={
+                'client_id': client_id,
+                'client_secret': self.client_secret,
+            },
+            auto_refresh_url='https://api.mollie.com/oauth2/tokens',
+            redirect_uri=redirect_uri,
+            scope=scope,
+            token=token,
+            token_updater=set_token
+        )
+        authorization_url = None
+        if not self.oauth.authorized:
+            authorization_url, state = self.oauth.authorization_url('https://www.mollie.com/oauth2/authorize')
```

Comment:
The url should be modifiable by implementors. Maybe add a class constant like `API_ENDPOINT`?

Reply:
Added ```
    OAUTH_AUTO_REFRESH_URL = API_ENDPOINT + '/oauth2/tokens'
    OAUTH_AUTHORIZATION_URL = API_ENDPOINT + '/oauth2/authorize'
    OAUTH_TOKEN_URL = API_ENDPOINT + '/oauth2/tokens'```